test/odbc/psqlodbc/test_pg_datatype_bigint.cpp

```diff
@@ -0,0 +1,1016 @@
+  rcode = SQLExecDirect(odbcHandler.GetStatementHandle(), (SQLCHAR *)UpdateTableStatement(TABLE_NAME, update_col, UPDATE_WHERE_CLAUSE).c_str(), SQL_NTS);
+  ASSERT_EQ(rcode, SQL_ERROR);
+  odbcHandler.CloseStmt();
+
+  // Assert that no values changed
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string>{COL1_NAME}));
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(pk_len, PK_BYTES_EXPECTED);
+  ASSERT_EQ(pk, PK_INSERTED);
+  ASSERT_EQ(data_len, DATA_BYTES_EXPECTED);
+  ASSERT_EQ(data, StringToBigInt(DATA_INSERTED));
+
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_BigInt, Arithmetic_Operators) {
+  vector<pair<string, string>> TABLE_COLUMNS = {
+    {COL1_NAME, DATATYPE_NAME + " PRIMARY KEY"},
+    {COL2_NAME, DATATYPE_NAME}
+  };
+
+  const int BYTES_EXPECTED = 8;
+
+  long long int pk;
+  long long int data;
+  SQLLEN pk_len;
+  SQLLEN data_len;
+  SQLLEN affected_rows;
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler(Drivers::GetDriver(ServerType::PSQL));
+
+  const vector<string> INSERTED_PK = {
+    "-73708",
+    "123",
+    "233"
+  };
+
+  const vector<string> INSERT_DATA = {
+    "2",
+    "3",
+    "5"
+  };
+  const int NUM_OF_DATA = INSERT_DATA.size();
+
+  const vector<string> OPERATIONS_QUERY = {
+    COL1_NAME + "+" + COL2_NAME,
+    COL1_NAME + "-" + COL2_NAME,
+    COL1_NAME + "/" + COL2_NAME,
+    COL1_NAME + "*" + COL2_NAME,
+
+    COL1_NAME + "^" + COL2_NAME, // Power
+    "|/" + COL2_NAME,            // Square Root
+    "||/" + COL2_NAME,           // Cube Root
+    "@" + COL1_NAME,             // Absolute Value
+  };
+  const int NUM_OF_OPERATIONS = OPERATIONS_QUERY.size();
+
+  vector<vector<long long int>> expected_results = {};
+
+  // initialization of expected_results
+  for (int i = 0; i < NUM_OF_DATA; i++) {
+    expected_results.push_back({});
+    long long int data_1 = StringToBigInt(INSERTED_PK[i]);
+    long long int data_2 = StringToBigInt(INSERT_DATA[i]);
+
+    expected_results[i].push_back(data_1 + data_2);
+    expected_results[i].push_back(data_1 - data_2);
+    expected_results[i].push_back(data_1 / data_2);
+    expected_results[i].push_back(data_1 * data_2);
+
+    expected_results[i].push_back(pow(data_1, data_2));
+    expected_results[i].push_back(sqrt(data_2));
+    expected_results[i].push_back(cbrt(data_2));
+    expected_results[i].push_back(abs(data_1));
+  }
+
+  long long int col_results[NUM_OF_OPERATIONS];
+  SQLLEN col_len[NUM_OF_OPERATIONS];
+  vector<tuple<int, int, SQLPOINTER, int, SQLLEN *>> BIND_COLUMNS = {};
+
+  // initialization for BIND_COLUMNS
+  for (int i = 0; i < NUM_OF_OPERATIONS; i++) {
+    tuple<int, int, SQLPOINTER, int, SQLLEN *> tuple_to_insert(i + 1, SQL_C_SBIGINT, (SQLPOINTER)&col_results[i], 0, &col_len[i]);
```

Comment:
Some of these operators in Postgres don't return type bigint (e.g., square root or cube root). Do we still want to bind it to SQL_C_SBIGINT? I assume that it would round it down to the nearest int, but we're also losing decimal places from these functions. 

if I were a user who wanted to know the square root of 2 and 2 was of type big int, I wouldn't be satisfied with the number 1.